cluster-autoscaler/core/scaledown/actuation/delete_in_batch.go

```diff
@@ -67,59 +67,55 @@ func NewNodeDeletionBatcher(ctx *context.AutoscalingContext, csr *clusterstate.C
 	}
 }
 
-// AddNode adds node to delete candidates and schedule deletion.
-func (d *NodeDeletionBatcher) AddNode(node *apiv1.Node, drain bool) error {
+// AddNodes adds node list to delete candidates and schedules deletion.
+func (d *NodeDeletionBatcher) AddNodes(nodes []*apiv1.Node, nodeGroup cloudprovider.NodeGroup, drain bool) {
 	// If delete interval is 0, than instantly start node deletion.
 	if d.deleteInterval == 0 {
-		nodeGroup, err := deleteNodesFromCloudProvider(d.ctx, []*apiv1.Node{node})
-		if err != nil {
-			result := status.NodeDeleteResult{ResultType: status.NodeDeleteErrorFailedToDelete, Err: err}
-			CleanUpAndRecordFailedScaleDownEvent(d.ctx, node, nodeGroup.Id(), drain, d.nodeDeletionTracker, "", result)
-		} else {
-			RegisterAndRecordSuccessfulScaleDownEvent(d.ctx, d.clusterState, node, nodeGroup, drain, d.nodeDeletionTracker)
+		err := deleteNodesFromCloudProvider(d.ctx, nodes, nodeGroup)
+		for _, node := range nodes {
+			if err != nil {
+				result := status.NodeDeleteResult{ResultType: status.NodeDeleteErrorFailedToDelete, Err: err}
+				CleanUpAndRecordFailedScaleDownEvent(d.ctx, node, nodeGroup.Id(), drain, d.nodeDeletionTracker, "", result)
+			} else {
+				RegisterAndRecordSuccessfulScaleDownEvent(d.ctx, d.clusterState, node, nodeGroup, drain, d.nodeDeletionTracker)
+			}
 		}
-		return nil
-	}
-	nodeGroupId, first, err := d.addNodeToBucket(node, drain)
-	if err != nil {
-		return err
+		return
 	}
+	first := d.addNodesToBucket(nodes, nodeGroup, drain)
 	if first {
-		go func(nodeGroupId string) {
+		go func(nodeGroup cloudprovider.NodeGroup) {
```

Comment:
Is `NodeGroup` always thread safe?

Reply:
I don't know, changed it back to string to avoid unwanted interactions.